app/models/subnet.rb

```diff
@@ -102,8 +102,10 @@ def model_name
   validates :nic_delay, numericality: { only_integer: true, greater_than_or_equal_to: 0, less_than: 1024 }, allow_blank: true
 
   before_validation :normalize_addresses
+  before_validation :normalize_subnet
   after_validation :validate_against_external_ipam
   validate :ensure_ip_addrs_valid
+  validate :ensure_gateway_in_subnet
```

Comment:
I don't think this should be in this PR. As I noted: it makes it more strict and isn't always true. Granted, it's an edge case but it at least deserves its own entry in the changelog and probably an upgrade warning. To get this PR merged, it's best left out of it.

Reply:
Yeah I misunderstood Evgeni above, you are right. Removed.